### DISCUSSION TITLE
Is there any way to create a custom content dialog?

### DISCUSSION THREAD

**Discussion by zengxs** [tauri-apps/tauri#4079] [Q&A] [answered]
I want to implement dialog like Apple's [`App-Modal Dialogs`](https://developer.apple.com/design/human-interface-guidelines/macos/windows-and-views/dialogs/), but tauri's doesn't seem to support custom content dialogs.

> By Apple's documentation, `App-Modal Dialogs` is <ins>**an app-modal dialog prevents the user from doing anything in the app until the dialog is dismissed.**</ins>

I think maybe this needs to be done using `window`, but I don't know how to do it.

Is there anyone who can share some experience with me or point me in some direction about this.
Thanks

**Comment by zengxs** [comment, marked as answer]
I was wrong, maybe I shouldn't have looked for a native dialog solution.

I found this very easy to implement in the web.

**Reply by FabianLars** [reply]
Since we don't offer a built-in way for "native" ones (and it should be relatively easy in html+js as you said), i'm gonna mark your comment as the answer.

**Comment by IsaacTru** [comment]
I'm not finding it as easy to implement. Using a `<dialog>` HTML element always shows in the UI, even before calling `showModal` or equivalent. This is not the behavior I expected from other resources. Is Tauri not handling this element properly or am I doing something wrong?

**Reply by YounesChellaf** [reply]
am having same issue, the dialog window opens up but the content isn't visible until i switch to another application then back to my main window where i see my modal content there
